src/Listener/ViewListener.php

```diff
@@ -245,17 +246,27 @@ protected function _getPageTitle()
     /**
      * Returns fields to be displayed on scaffolded template
      *
+     * @param array $assocations Associations list.
      * @return array
      */
-    protected function _scaffoldFields()
+    protected function _scaffoldFields(array $associations = [])
     {
-        $cols = $this->_table()->schema()->columns();
-        $scaffoldFields = array_combine(array_values($cols), array_fill(0, count($cols), []));
-
         $action = $this->_action();
         $configuredFields = $action->config('scaffold.fields');
         if (!empty($configuredFields)) {
             $scaffoldFields = Hash::normalize($configuredFields);
+        } else {
+            $cols = $this->_table()->schema()->columns();
+            $scaffoldFields = array_combine(array_values($cols), array_fill(0, count($cols), []));
```

Comment:
@lorenzo `Hash::normalize($cols)` generates array like `['field' => null]` instead of `['field' => []]` by current statement. But further down the fields are looped over and value cast to array so i'll change to using `Hash::normalize()`